src/functional/ffront/common_types.py

```diff
@@ -59,8 +59,8 @@ class DimensionType(SymbolType):
 
 @dataclass(frozen=True)
 class OffsetType(SymbolType):
-    source: Optional[func_common.Dimension] = None
-    target: Optional[tuple[func_common.Dimension, ...]] = None
+    source: func_common.Dimension
+    target: tuple[func_common.Dimension] | tuple[func_common.Dimension, func_common.Dimension]
```

Comment:
Maybe I am missing something, but why do you have a tuple with only one argument?

Reply:
gotcha